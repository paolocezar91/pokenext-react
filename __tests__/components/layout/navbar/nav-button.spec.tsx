import NavButton from "@/components/layout/navbar/nav-button";
import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";

describe("NavButton Component", () => {
  it("renders correctly with name and class", () => {
    render(<NavButton className="bulbasaur">Bulbasaur</NavButton>);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent("Bulbasaur");
    expect(button).toHaveClass("bulbasaur");
  });

  it("click callback and onMouseEnterCallback are fired", () => {
    const clickCallback = jest.fn();
    const onMouseEnterCallback = jest.fn();

    render(
      <NavButton
        onMouseEnter={() => onMouseEnterCallback()}
        onClick={() => clickCallback()}
        className="bulbasaur"
      >
        Bulbasaur
      </NavButton>,
    );
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(clickCallback).toHaveBeenCalled();
    fireEvent.mouseEnter(button);
    expect(onMouseEnterCallback).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<NavButton>Bulbasaur</NavButton>);
    expect(asFragment()).toMatchSnapshot();
  });
});
