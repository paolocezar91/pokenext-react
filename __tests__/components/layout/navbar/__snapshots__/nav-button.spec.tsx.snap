// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton Component should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="
    flex
    px-2
    py-2
    text-sm
    rounded
    cursor-pointer
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark)
   bg-transparent  btn btn-primary"
    type="button"
  >
    Bulbasaur
  </button>
</DocumentFragment>
`;
